### ISSUE TITLE
Enforce random seed in stochastic learners

### ISSUE THREAD

**Issue by bfolie** [CitrineInformatics/smlb#7]
The skleran learners expose a random state seed but it is not required. [Example](https://github.com/CitrineInformatics/smlb/blob/master/smlb/learners/scikit_learn/gaussian_process_regression_sklearn.py). In keeping with smlb's focus on reproducibility, these classes should extend [Random](https://github.com/CitrineInformatics/smlb/blob/master/smlb/core/random.py#L66)

**Comment by bfolie** [comment]
Addressed in #12
